src/initialize_dae.jl

```diff
@@ -54,19 +64,32 @@ end
 
 function _initialize_dae!(integrator, prob::ODEProblem,
         alg::DefaultInit, x::Val{true})
-    _initialize_dae!(integrator, prob,
-        BrownFullBasicInit(integrator.opts.abstol), x)
+    if SciMLBase.has_initializeprob(prob.f)
+        _initialize_dae!(integrator, prob,
+            OverrideInit(integrator.opts.abstol), x)
+    else
+        _initialize_dae!(integrator, prob,
+            BrownFullBasicInit(integrator.opts.abstol), x)
+    end
 end
 
 function _initialize_dae!(integrator, prob::ODEProblem,
         alg::DefaultInit, x::Val{false})
-    _initialize_dae!(integrator, prob,
-        BrownFullBasicInit(integrator.opts.abstol), x)
+    if SciMLBase.has_initializeprob(prob.f)
+        _initialize_dae!(integrator, prob,
+            OverrideInit(integrator.opts.abstol), x)
+    else
+        _initialize_dae!(integrator, prob,
+            BrownFullBasicInit(integrator.opts.abstol), x)
+    end
```

Comment:
can't these be combined?

Reply:
maybe, since it wasn't already I assumed that it's potentially due to some ambiguity and with such a big change going on I don't want to touch that at the same time.